### PR TITLE
docs: add a few more k6 browser examples

### PR DESCRIPTION
**Pull request by mdcruz**
## What?

This PR adds a few more examples of using k6 browser, mainly using custom metrics and hybrid performance. 

## Checklist

Please fill in this template:
- [x] I have used a meaningful title for the PR.
- [x] I have described the changes I've made in the "What?" section above.
- [x] I have performed a self-review of my changes.
- [x] I have run the `make docs` command locally and verified that the changes look good.

Select one of these and delete the others:

If updating the documentation for the most recent release of k6: 
- [x] I have made my changes in the `docs/sources/v{most_recent_release}` folder of the documentation.
- [x] I have reflected my changes in the `docs/sources/next` folder of the documentation.
- [x] I have reflected my changes in the relevant (*e.g.*  when correcting a documentation error) folders of the previous k6 versions of the documentation.

If updating the documentation for the next release of k6:
- [ ] I have made my changes in the `docs/sources/next` folder of the documentation.

### CODE REVIEW

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+title: 'Hybrid approach to performance'
+heading: 'Hybrid performance with k6 browser'
+head_title: 'Hybrid performance with k6 browser'
+excerpt: 'An example on how to implement a hybrid approach to performance with k6 browser'
+weight: 01
+---
+
+# Hybrid performance with k6 browser
+
+An alternative approach to [browser-based load testing](https://grafana.com/docs/k6/<K6_VERSION>/testing-guides/load-testing-websites/#browser-based-load-testing) that is much less resource-intensive is combining a low amount of virtual users for a browser test with a high amount of virtual users for a protocol-level test.
```
Comment (inancgumus) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:11]: Is `<K6 VERSION>` here correct?

**Review by inancgumus — COMMENTED** [review]
Looks like a great work :) I haven't looked it entirely, but one question.

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md (hunk 1 above)
Comment (mdcruz) [reply docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:11]: Yes, I've looked at the existing examples and seems to be correct :)

**Review by ankur22 — APPROVED** [review]
LGTM

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md (hunk 1 above)
Comment (heitortsergent) [reply docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:11]: That's correct! It's how we handle versioned links in the Grafana docs, it'll change based on which version of the docs the user is in. So if they're seeing `latest`, the links on that page will replace `<K6_version>` with `latest`.

File: docs/sources/next/using-k6-browser/metrics.md [hunk 2]
```diff
@@ -98,3 +98,75 @@ When the test is run, you should see a similar output as the one below.
...
   browser_web_vital_ttfb.........................: avg=339.3ms  min=258.9ms  med=339.3ms  max=419.7ms  p(90)=403.62ms p(95)=411.66ms
 ```
+
+## Measure custom metrics
+
+Through k6 browser's `page.evaluate` function, you can call the [Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance_API) to measure the performance of web applications. As an example, if you want to measure the time it takes for your users to complete actions, such as a search feature, then you can use the [`performance.mark`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) method to add a timestamp in your browser's performance timeline. 
+
+To measure the time difference between two performance markers, you can use the [`performance.measure`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/measure) method.
+
+The time duration that is returned by the performance measure can be added as custom metric in k6 browser using [Trends](https://k6.io/docs/javascript-api/k6-metrics/trend/).
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/metrics.md:108]:
```suggestion
When using the k6 browser `page.evaluate` function, you can call the [Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance_API) to measure the performance of web applications. For example, if you want to measure the time it takes for your users to complete actions, such as a search feature, you can use the [`performance.mark`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) method to add a timestamp in your browser's performance timeline. 
		
Using the [`performance.measure`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/measure) method, you can also measure the time difference between two performance markers. The time duration that `performance.measure` returns can be added as a custom metric in k6 browser using [Trends](https://k6.io/docs/javascript-api/k6-metrics/trend/).
```

File: docs/sources/next/using-k6-browser/metrics.md [hunk 3]
```diff
@@ -98,3 +98,75 @@ When the test is run, you should see a similar output as the one below.
...
+    myTrend.add(total_action_time);
+  } finally {
+    page.close();
+  }
+}
+```
+
+{{< /code >}}
+
+When the test is run, you should see a similar output as the one below.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/metrics.md:166]:
```suggestion
After you run the test, you should see a similar output as the one below.
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md (hunk 1 above)
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:11]:
```suggestion
An alternative approach to [browser-based load testing](https://grafana.com/docs/k6/<K6_VERSION>/testing-guides/load-testing-websites/#browser-based-load-testing) that's much less resource-intensive is combining a small number of virtual users for a browser test with a large number of virtual users for a protocol-level test. 
```

Small nit, I think this sounds a bit more natural when referring to "users".

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+head_title: 'Hybrid performance with k6 browser'
+excerpt: 'An example on how to implement a hybrid approach to performance with k6 browser'
+weight: 01
+---
+
+# Hybrid performance with k6 browser
+
+An alternative approach to [browser-based load testing](https://grafana.com/docs/k6/<K6_VERSION>/testing-guides/load-testing-websites/#browser-based-load-testing) that is much less resource-intensive is combining a low amount of virtual users for a browser test with a high amount of virtual users for a protocol-level test. 
+
+Hybrid performance can be achieved in multiple ways, often using different tools. To simplify the developer experience, k6 browser can be easily combined with core k6 features, so you can easily write hybrid tests in a single script.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:13]:
```suggestion
You can achieve hybrid performance in multiple ways, often by using different tools. To simplify the developer experience, you can combine k6 browser with core k6 features to write hybrid tests in a single script. 
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+# Hybrid performance with k6 browser
+
+An alternative approach to [browser-based load testing](https://grafana.com/docs/k6/<K6_VERSION>/testing-guides/load-testing-websites/#browser-based-load-testing) that is much less resource-intensive is combining a low amount of virtual users for a browser test with a high amount of virtual users for a protocol-level test. 
+
+Hybrid performance can be achieved in multiple ways, often using different tools. To simplify the developer experience, k6 browser can be easily combined with core k6 features, so you can easily write hybrid tests in a single script. 
+
+## Browser and HTTP test
+
+The code below shows an example of combining a browser and HTTP test in a single script. While the backend is exposed to the typical load, the frontend is also checked for any unexpected issues. Thresholds are defined to check both HTTP and browser metrics against pre-defined SLOs.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:17]:
```suggestion
The code below shows an example of combining a browser and HTTP test in a single script. While the script exposes the backend to the typical load, it also checks the frontend for any unexpected issues. It also defines thresholds to check both HTTP and browser metrics against pre-defined SLOs.
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  }
+}
+
+```
+
+{{< /code >}}
+
+## Browser and failure injection test
+
+A browser test can also be run with a failure injection test via the [xk6-disruptor](https://github.com/grafana/xk6-disruptor) extension. This approach lets you find issues in your front end if any services it depends on are suddenly injected with failures, such as delays or server errors.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:109]:
```suggestion
You can also run a browser test together with a failure injection test by using the [xk6-disruptor](https://github.com/grafana/xk6-disruptor) extension. This approach lets you find issues in your front end if any services it depends on are suddenly injected with failures, such as delays or server errors.
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+The code below shows an example of introducing faults to a Kubernetes service. While this happens, the `browser` scenario is also executed, which checks the frontend application for unexpected errors that might not have been handled properly.
+
+To find out more information about injecting faults to your service, check out our [get started guide with xk6-disruptor](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/xk6-disruptor/get-started/).
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:113]:
```suggestion
To find out more information about injecting faults to your service, check out the [Get started with xk6-disruptor guide](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/xk6-disruptor/get-started/).
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+  }
+}
+
+```
+
+{{< /code >}}
+
+## Recommended practices
+
+- **Do start small**. Start with a low amount of browser-based virtual users. A good starting point is to have 10% virtual users or less to monitor the user experience for your end-users, while around 90% of traffic should be emulated from the protocol level.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:195]:
```suggestion
- **Start small**. Start with a small number of browser-based virtual users. A good starting point is to have 10% virtual users or less to monitor the user experience for your end-users, while the script emulates around 90% of traffic from the protocol level.
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+- **Combine browser test with different load testing types**. To fully understand the impact of different traffic patterns on your end-user experience, experiment with running your browser test with different [load testing types](https://grafana.com/docs/k6/<K6_VERSION>/testing-guides/test-types/).
+- **Cover high-risk user journeys as a start**. Consider identifying the high-risk user journeys first so you can start to monitor the web performance metrics for these user journeys while your backend applications are being exposed to high traffic or service faults.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:197]:
```suggestion
- **Focus on high-risk user journeys as a start**. Identify the high-risk user journeys first so you can start monitoring the web performance metrics for them while your backend applications are being exposed to high traffic or service faults.
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+```
+
+{{< /code >}}
+
+## Browser and failure injection test
+
+A browser test can also be run with a failure injection test via the [xk6-disruptor](https://github.com/grafana/xk6-disruptor) extension. This approach lets you find issues in your front end if any services it depends on are suddenly injected with failures, such as delays or server errors.
+
+The code below shows an example of introducing faults to a Kubernetes service. While this happens, the `browser` scenario is also executed, which checks the frontend application for unexpected errors that might not have been handled properly.
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:111]:
```suggestion
The following code shows an example of how to introduce faults to a Kubernetes service. At the same time, the `browser` scenario runs to ensure the frontend application is free of any unexpected errors that may not have been handled properly.
```

**Review by heitortsergent — COMMENTED** [review]
Looks good overall! I made a few comments most of them are just to rewrite a few sentences to use [active voice](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#write-in-active-voice) instead of passive to follow our style guide. 🤓

**Review by inancgumus — APPROVED** [review]

**Pull request merged**